Handle action db and model db different

dpsutton opened the pull request:
The +New -> Action workflow lets you create an action. This starts with the user choosing the database, writing the action. On save it prompts you to choose a model it should live under. But this model's db and the db chosen originally are not checked in the FE to be the same. This is also desired behavior.

When executing, we check that the action's database has actions enabled. But "action's database" is ambiguous now: the model's database or the query action's database. This PR ensures we check both when they are different.

> **Warning**
> The screenshot is out of date. The error visible there is that the databases don't agree. That is now fine behavior, but they both must have actions enabled.

#### The Workflow
When clicking on "+New" and choosing "Action", you just dive in with a database selector and start writing sql.
<img width="1182" alt="image" src="https://user-images.githubusercontent.com/6377293/221265051-5bd36e03-045e-4745-8b8a-157288052f32.png">

But you can start writing your action against whatever table you like. Only on saving do you have to attach your action to a model. But the models are displayed by collection and give no idea about the underlying database, or if they have actions enabled.

So you can easily add an action on database 1 to a model on database 2.
![image](https://user-images.githubusercontent.com/6377293/221266741-af1e12bb-e0d7-4575-b43e-faa2d7f17a74.png)


<img width="1182" alt="image" src="https://user-images.githubusercontent.com/6377293/221264660-15efef22-7f87-4824-99dc-4ec08502f18b.png">

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/28616)
<!-- Reviewable:end -->

dpsutton committed:
Ensure query action's db matches model id

This is only a half-way measure. Needs some FE loving as well.

The +New -> Action workflow lets you create an action. This starts with
the user choosing the database, writing the action. On save it prompts
you to choose a model it should live under. But this model's db and the
db chosen originally are not checked in the FE.

Ideally we only show models on that existing db, but it kinda seems
weird to have an action do things to a random table and it live under a
model that just needs to be on the db.

mazameli commented on src/metabase/api/action.clj:
```diff
@@ -104,6 +104,13 @@
+    (when (and database_id (not= database_id (:database_id model)))
+      (let [model-db (db/select-one Database :id (:database_id model))
+            action-db (db/select-one Database :id database_id)]
+       (throw (ex-info (tru "Model database not equal to action database")
```
```suggestion
       (throw (ex-info (tru "Actions need to use the same database as the model.")
```

dpsutton committed:
reword error message

Co-authored-by: Maz Ameli <maz@metabase.com>

dpsutton committed:
Check both databases for enabled actions when they disagree

a custom action can have its own query which brings its own
database. When creating a new custom action from the models detail page,
these go hand in hand. When using the workflow from the "+New" button,
the model is chosen at the last step, well after the action's db and
query is set. And there's nothing holding these two together.

Asked if this was a bug and was told no. But we want to make sure we
check both for actions enabled.

dpsutton committed:
fix lint

tsmacdonald reviewed (approved):
:boat:
